### PR TITLE
Use error message from mailerlite as error text

### PR DESCRIPTION
**Pull request by ludeeus**

### CODE REVIEW

File: src/services/newsletter.ts [hunk 1]
```diff
@@ -50,8 +50,10 @@ export async function newsletterHandler(
   });
 
   if (!response.ok) {
+    const data = await response.json<Record<string, any>>();
```
Comment (balloob) [inline src/services/newsletter.ts:53]: this can raise in case Mailterlite broke so much that it didn't even respond with JSON.

Comment (ludeeus) [reply src/services/newsletter.ts:53]: If that would happen we catch that on the router 👍

Comment (balloob) [reply src/services/newsletter.ts:53]: Do you still use `NewsletterErrorType.SUBSCRIPTION` then ?

Comment (ludeeus) [reply src/services/newsletter.ts:53]: No :shrug:

File: src/services/newsletter.ts [hunk 2]
```diff
@@ -40,22 +40,42 @@ export async function newsletterHandler(
...
       NewsletterErrorType.SUBSCRIPTION
     );
   }
 
+  if (!response.ok) {
+    try {
+      data = await response.json<Record<string, any>>();
+    } catch (err: any) {
+      throw new ServiceError(
+        "Could not parse error",
```
Comment (balloob) [inline src/services/newsletter.ts:67]: We still need to tell user that their action went wrong
```suggestion
        "Could not subscribe (unknown error)",
```

**Pull request merged**